CMSgov/easi-app
# Cleanup of TRB store code for consistency

No associated Jira ticket - doing some cleanup of TRB code for consistency, after talking with Clay and Jeremy on 1/5, as well as fixing a couple of potential issues I noticed when implementing and testing.

## Changes and Description

- Change TRB store methods to take a full `context.Context` instead of just a `*zap.Logger`.
- Use consistent parameter names in the resolvers and store methods for TRB request documents. The main functional change is using `"trbRequestID"` as the key for structured logging, like all other TRB store code does, instead of `"requestID"`.
- Added logging in the error-handling blocks following some `PrepareNamed()` calls in TRB store methods. For some types of errors (such as an inability to connect to the database), `PrepareNamed()` returns an error before a `Get()`/`Select()` call is ever made; this PR makes sure those errors are logged for all TRB store methods.
- Changed `pkg/storage/trb_advice_letter.go` to correctly update `modified_by`, `modified_at`.

## How to test this change

CI tests running successfully should be sufficient; if you really want to go the extra mile, you can trigger errors (by messing with the SQL queries or stopping the database container) and make sure that errors are logged.

For testing the modified_by/modified_at changes, follow the tests in https://github.com/CMSgov/easi-app/pull/1865, and verify that those fields get updated. Using a different EUA ID in the local authentication header when calling `requestReviewForTRBAdviceLetter` and `sendTRBAdviceLetter` will be reflected in the `modifiedBy` field.
